### ISSUE TITLE
False positive?: get_unwrap

### ISSUE THREAD

**Issue by Thinkofname** [Manishearth/rust-clippy#1725]
Link to example: http://play.integer32.com/?gist=2d02b966174279055a9519a5f3182a7d&version=undefined

Honestly seems like the suggested change should work but lifetimes seem to prevent it. Might be because `Index` and `get` take slightly different parameter types?

**Comment by oli-obk** [comment]
that.... is ugly. I'm almost tempted to call it a stdlib bug, but there's probably some reasoning behind it.

get is

```rust
fn get<Q: ?Sized>(&self, k: &Q) -> Option<&V> where K: Borrow<Q>, Q: Hash + Eq
```

while Index is

```rust
impl<'a, K, Q: ?Sized, V, S> Index<&'a Q> for HashMap<K, V, S> where K: Eq + Hash + Borrow<Q>,
        Q: Eq + Hash,
        S: BuildHasher
type Output = V
fn index(&self, index: &Q) -> &V
```

> Might be because Index and get take slightly different parameter types?

The only difference I see is the `'a` lifetime, which can't really be it. Probably something todo with the indexing desugaring.

Separate bug: the suggestion is wrong (the first ampersand is not correct here)

**Comment by Thinkofname** [comment]
Looks like its this rustc issue https://github.com/rust-lang/rust/issues/32382

**Comment by JelteF** [comment]
I'm having this same issue on this line: 
https://github.com/JelteF/derive_more/blob/2e37e7f8181ba333df9503088222772b77c83d91/src/from.rs#L103

I'm working around this by manually calling `index()`.

**Comment by giraffate** [comment]
Did the `#[warn(clippy::get_unwrap)]` be added? The [`get_unwrap`](https://rust-lang.github.io/rust-clippy/master/index.html#get_unwrap) is allow-by-default now.
